### PR TITLE
USFIMR S1 MLData Catalog

### LINKED ISSUE #57: USFIMR, S1 ml-aoi catalog
Create ml-aoi STAC catalog that will represent the setup for model training

USFIMR has 48 flood events. We need to choose some subset as test and validation.
We can either choose entire flood events or some geographic subset of events.
This decision will likely require manual inspection of events to determine what is and is not representative 
/cc @jamesmcclain -- your call on the strategy on this one

Feature: 
* https://github.com/azavea/noaa-flood-mapping/issues/56
* https://github.com/azavea/noaa-flood-mapping/issues/3

Label: 
* https://github.com/azavea/noaa-flood-mapping/issues/9

Output:
* ml-aoi catalog with S1 feature and USFIMR labels
* ml-aoi catalog with S1, HAND features and USFIMR labels

Note:
This should be a chance to use stac-packer to generate the catalog with minimum fuss.

### PR DESCRIPTION
**Pull request by CloudNiner**
# Overview

Generates an mldata catalog of the same structure as `./catalogs/sen1floods11-mldata` but for the usfimr-s1 catalog generated in `./catalogs/usfimr-s1`. 

This PR remains in draft form because I'm awaiting guidance on how to do the test, train, validation splits. Otherwise, we could write a new rv_stac_config over in `./inference` with what's here and run a RV prediction, even if it does generate nonsense based on the current test, train, validation splits.

Here's the generated catalog for your perusal:

[usfimr-s1-mldata-catalog.zip](https://github.com/azavea/noaa-flood-mapping/files/5211109/usfimr-s1-mldata-catalog.zip)

Closes #57

### CODE REVIEW

**Comment by moradology** [comment]
A randomized set (per discussion with @jamesmcclain and @echeipesh) has been uploaded to s3://usfimr-s1-mldata/usfimr-s1-mldata-catalog_seed42.tar.gz

File: catalogs/usfimr-s1/build_mldata_catalog.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import argparse
+
+import pystac
+
+import numpy as np
+from stac_utils.s3_io import register_s3_io
+from sklearn.model_selection import train_test_split
```
Comment (CloudNiner) [inline catalogs/usfimr-s1/build_mldata_catalog.py:7]: I don't think scikit learn is in the requirements.txt for the parent docker container in ./catalogs. If it isn't don't forget to add it along with any other reps you need for this PR

**Review by echeipesh — APPROVED** [review]
